src/basic/terminal-util.c

```diff
@@ -2427,6 +2445,158 @@ int terminal_get_size_by_dsr(
         return r;
 }
 
+#define MAX_TERMINFO_LENGTH 64
+/* python -c 'print("".join(hex(ord(i))[2:] for i in "TN").upper())' */
+#define DCS_TERMINFO_Q ANSI_DCS "+q" "544E" ANSI_ST
+/* The answer is either 0+r… (invalid) or 1+r… (OK). */
+#define DCS_TERMINFO_R0 ANSI_DCS "0+r" "544E" ANSI_ST
```

Comment:
Technically the correct response for an unrecognised cap should be `DCS 0 + r ST` without the actual cap name, but in practice some terminals include the name and some don't. I'm not sure if this actually makes any difference to this code, but just FYI.

Reply:
Yeah, we are prepared. We have both a timeout and we check the received reply for correctness. (Unless we get a syntactically reply with wrong content. Can't help that.)